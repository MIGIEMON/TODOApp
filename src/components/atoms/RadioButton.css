.Radio {
    width: 130px;
    text-align: center;
    padding-right: 10px;
}

.Radio input[type='radio'] {
    position: absolute;
    z-index: -1;
    opacity: 0;
}

.Radio label {
    position: relative;
    display: inline-block;
    font-size: 15px;
    padding-right: 8px;
    padding-left: 30px;
    cursor: pointer;
    color: #a6a6a6;
}

.Radio label::before {
    position: absolute;
    z-index: -1;
    top: -1.5px;
    left: 0;
    display: block;
    width: 19px;
    height: 19px;
    content: ' ';
    border: 3px solid #a6a6a6;
    border-radius: 4px;
    align-items: center;
}

.Radio input[type='radio']+label::before {
    border-radius: 18px;
    align-items: center;
}

.Radio input[type='radio']:checked+label {
    padding-right: 8px;
    padding-left: 30px;
    color: #ffffff;
}

.Radio input[type='radio']:checked+label::before {
    top: -2.5px;
    left: 8px;
    width: 100%;
    height: 100%;
    border: 3px solid #d27171;
    background-color: #d27171;
}

/* Transition */
.Radio label,
.Radio label::before {
    -webkit-transition: 0.25s all ease;
    transition: 0.25s all ease;
}