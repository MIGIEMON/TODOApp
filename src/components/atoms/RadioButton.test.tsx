import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { RadioButton } from './RadioButton';

describe('RadioButtonコンポーネント', () => {
  const onChangeMock = jest.fn();
  const props = {
    radioButtonType: 'red',
    checked: false,
    onChange: onChangeMock,
    id: 'radioId',
    name: 'radioName'
  };

  it('ラジオボタンが表示される', () => {
    const { container } = render(<RadioButton {...props}>テスト用ラベル</RadioButton>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('ラジオボタンをクリックするとonChange関数が呼び出される', () => {
    const { getByLabelText } = render(<RadioButton {...props}>テスト用ラベル</RadioButton>);
    fireEvent.click(getByLabelText('テスト用ラベル'));
    expect(onChangeMock).toHaveBeenCalledTimes(1);
  });

  it('ラジオボタンが選択される', () => {
    const { getByLabelText } = render(<RadioButton {...props} checked={true}>テスト用ラベル</RadioButton>);
    expect(getByLabelText('テスト用ラベル')).toBeChecked();
  });
});
