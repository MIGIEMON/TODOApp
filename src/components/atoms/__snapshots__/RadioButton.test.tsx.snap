// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButtonコンポーネント ラジオボタンが表示される 1`] = `
<div
  class="Radio red"
>
  <input
    id="radioId"
    name="radioName"
    type="radio"
  />
  <label
    for="radioId"
  >
    テスト用ラベル
  </label>
</div>
`;
